### ISSUE TITLE
xwebadministration dsc_xiislogging

### ISSUE THREAD

**Issue by epsilon-jpage** [puppetlabs/ruby-pwsh#54]
## Describe the Bug
When trying to use the dsc_xiislogging class to Add IIS X-Forwarded-For Log File Flag to the IIS Server default it is not working as expecgted..

## Expected Behavior
Custom IIS X-Forwarded-For Log File Flag should be configured to the IIS server.

## Steps to Reproduce
In the Puppetfile I added ...
mod 'dsc-xwebadministration', '3.2.0-0-0'

As per documentation found [here](https://github.com/dsccommunity/xWebAdministration/blob/d66aeeb841c6dad56bf17ba0abffc5ca598c1bbb/tests/Integration/MSFT_XIISLogging.config.ps1), in a test profile I wrote ...
```
  dsc_xiislogging { "Add IIS X-Forwarded-For Log File Flag":
    dsc_ensure => "Present",
    dsc_logcustomfields => "@(
    MSFT_xLogCustomField
      {
        LogFieldName = 'X-Forwarded-For'
        SourceName   = 'X-Forwarded-For'
        SourceType   = 'RequestHeader'
      }
    )",
  }
```

Upon puppet agent -t run ...
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: no parameter named 'dsc_ensure' (file: /etc/puppetlabs/code/environments/development/site-modules/profile/manifests/test_webserver_profile.pp, line: 50) on Dsc_xiislogging[Add IIS X-Forwarded-For Log File Flag] (file: /etc/puppetlabs/code/environments/development/site-modules/profile/manifests/test_webserver_profile.pp, line: 50) on node dw-hwke-web01.aws.cloud.epsilon
Warning: Not using cache on failed catalog

Error: Could not retrieve catalog; skipping run

## Environment
Windows Server 2019.
Puppet Enterprise 2019.8.1
dsc-xwebadministration 3.2.0-0-0

**Comment by michaeltlombardi** [comment]
This DSC resource does not have a `dsc_ensure` key - every resource currently has `ensure` to allow Puppet to do automagic comparisons (and defaults to `present`, meaning you do not have to specify it) - try the same definition sans `dsc_ensure`.

**Comment by michaeltlombardi** [comment]
Also, this issue is more appropriate in puppetlabs/Puppet.Dsc, which is the PowerShell module that builds the types; this repository is for issues with the underlying implementation in ruby, ie the passing back and forth of parameters to `Invoke-Dsc` and the output from that command back to ruby.

**Comment by epsilon-jpage** [comment]
I have created Bug Issue 73 in puppetlabs/Puppet.Dsc.

I also added to it and now to here ...
When I try this configuration ...
```
  dsc_xiislogging { "Add IIS X-Forwarded-For Log File Flag":
    dsc_logcustomfields => "@(
    MSFT_xLogCustomField
      {
        LogFieldName = 'X-Forwarded-For'
        SourceName   = 'X-Forwarded-For'
        SourceType   = 'RequestHeader'
      }
    )",
  }
```
I get ...
Error: Could not autoload puppet/type/dsc_xiislogging: Illegal Struct member key type. Expected NotUndef, Optional, String, or Enum. Got: Puppet::Pops::Types::PTypeReferenceType

Error: Failed to apply catalog: Could not autoload puppet/type/dsc_xiislogging: Illegal Struct member key type. Expected NotUndef, Optional, String, or Enum. Got: Puppet::Pops::Types::PTypeReferenceType

When I try this config ...
```
  dsc_xiislogging { "Add IIS X-Forwarded-For Log File Flag":
    ensure => "Present",
    dsc_logcustomfields => "@(
    MSFT_xLogCustomField
      {
        LogFieldName = 'X-Forwarded-For'
        SourceName   = 'X-Forwarded-For'
        SourceType   = 'RequestHeader'
      }
    )",
  }
```
I also get ...
Error: Could not autoload puppet/type/dsc_xiislogging: Illegal Struct member key type. Expected NotUndef, Optional, String, or Enum. Got: Puppet::Pops::Types::PTypeReferenceType

Error: Failed to apply catalog: Could not autoload puppet/type/dsc_xiislogging: Illegal Struct member key type. Expected NotUndef, Optional, String, or Enum. Got: Puppet::Pops::Types::PTypeReferenceType

Feel free to close if the other repo is the correct one.
